pybamm/solvers/jax_solver.py

```diff
@@ -0,0 +1,189 @@
+        'BDF' uses custom jax_bdf_integrate (see jax_bdf_integrate.py for details)
+    rtol : float, optional
+        The relative tolerance for the solver (default is 1e-6).
+    atol : float, optional
+        The absolute tolerance for the solver (default is 1e-6).
+    extra_options : dict, optional
+        Any options to pass to the solver.
+        Please consult `JAX documentation
+        <https://github.com/google/jax/blob/master/jax/experimental/ode.py>`_
+        for details.
+    """
+
+    def __init__(self, method='RK45', rtol=1e-6, atol=1e-6, extra_options=None):
+        super().__init__(method, rtol, atol)
+        self.ode_solver = True
+        method_options = ['RK45', 'BDF']
+        if method not in method_options:
+            raise ValueError('method must be one of {}'.format(method_options))
+        self.extra_options = extra_options or {}
+        self.name = "JAX solver"
+        self._cached_solves = dict()
+
+    def get_solve(self, model, t_eval):
+        """
+        Return a compiled JAX function that solves an ode model with input arguments.
+
+        Parameters
+        ----------
+        model : :class:`pybamm.BaseModel`
+            The model whose solution to calculate.
+        t_eval : :class:`numpy.array`, size (k,)
+            The times at which to compute the solution
+
+        Returns
+        -------
+        function
+            A function with signature `f(inputs)`, where inputs are a dict containing
+            any input parameters to pass to the model when solving
+
+        """
+        if model not in self._cached_solves:
+            if model not in self.models_set_up:
+                raise RuntimeError("Model is not set up for solving, run"
+                                   "`solver.solve(model)` first")
+
+            self._cached_solves[model] = self._create_solve(model, t_eval)
+
+        return self._cached_solves[model]
+
+    def _create_solve(self, model, t_eval):
+        """
+        Return a compiled JAX function that solves an ode model with input arguments.
+
+        Parameters
+        ----------
+        model : :class:`pybamm.BaseModel`
+            The model whose solution to calculate.
+        t_eval : :class:`numpy.array`, size (k,)
+            The times at which to compute the solution
+
+        Returns
+        -------
+        function
+            A function with signature `f(inputs)`, where inputs are a dict containing
+            any input parameters to pass to the model when solving
+
+        """
+        if model.convert_to_format != "jax":
+            raise RuntimeError("Model must be converted to JAX to use this solver"
+                               " (i.e. `model.convert_to_format = 'jax')")
+
+        if model.terminate_events_eval:
+            raise RuntimeError("Terminate events not supported for this solver."
+                               " Model has the following events:"
+                               " {}".format(model.events))
```

Comment:
Can you add some info to explain how to remove events from the model (`model.events = []`)? And perhaps suggest solving first with a different solver to figure out what the end time should be, then using this one for multiple fast iterations

Reply:
Yup, good idea. I've added a comment to this effect